daprdocs/content/en/developing-applications/building-blocks/state-management/howto-outbox.md

```diff
@@ -106,6 +108,73 @@ spec:
     value: "newOrder"
 ```
 
+### Shape the outbox pattern message
+
+You can override the outbox pattern message saved to the database during the transaction by setting a different message. This is done via a projected transaction payload, which is ignored in the database, but used as the outbox pattern message published to the user topic.
+
+In the following Go example of a state transaction, the value of `"2"` is saved to the database, but the value of `"3"` is published to the end-user topic.
+
+```go
+ops := make([]*dapr.StateOperation, 0)
+
+op1 := &dapr.StateOperation{
+    Type: dapr.StateOperationTypeUpsert,
+    Item: &dapr.SetStateItem{
+        Key:   "key1",
+        Value: []byte("2"),
+    },
+}
+op2 := &dapr.StateOperation{
+    Type: dapr.StateOperationTypeUpsert,
+    Item: &dapr.SetStateItem{
+        Key:   "key1",
+				Value: []byte("3"),
+         // Override the data payload saved to the database 
+				Metadata: map[string]string{
+					"outbox.projection": "true",
+        },
+    },
+}
+ops = append(ops, op1, op2)
+meta := map[string]string{}
+err := testClient.ExecuteStateTransaction(ctx, store, meta, ops)
+```
+
+By setting the metadata item `"outbox.projection"`  to `"true"`, the transaction value saved to the database is ignored, while the second value is published to the configured pub/sub topic. 
+
+### Override Dapr-generated cloudevent fields
+
+You can also override the [Dapr-generated cloudevent fields]({{< ref "pubsub-cloudevents.md#dapr-generated-cloudevents-example" >}}) on the published outbox event with custom cloudevent metadata.
+
+```go
+ops := make([]*dapr.StateOperation, 0)
+
+op1 := &dapr.StateOperation{
+    Type: dapr.StateOperationTypeUpsert,
+    Item: &dapr.SetStateItem{
+        Key:   "key1",
+        Value: []byte("2"),
+        // Override the data payload saved to the database 
+        Metadata: map[string]string{
+            "outbox.projection": "true",
```

Comment:
suggest to remove the `outbox.projection` item here for brevity, also we are not using the `outbox.cloudevent.*` pattern. so `outbox.cloudevent.id` should be `id`